libvmaf/src/fex_ctx_vector.c

```diff
@@ -33,19 +33,52 @@ int feature_extractor_vector_init(RegisteredFeatureExtractors *rfe)
 }
 
 int feature_extractor_vector_append(RegisteredFeatureExtractors *rfe,
-                                    VmafFeatureExtractorContext *fex_ctx)
+                                    VmafFeatureExtractorContext *fex_ctx,
+                                    uint64_t flags)
 {
     if (!rfe) return -EINVAL;
     if (!fex_ctx) return -EINVAL;
 
     for (unsigned i = 0; i < rfe->cnt; i++) {
         if (!strcmp(rfe->fex_ctx[i]->fex->name, fex_ctx->fex->name)) {
-            VmafDictionary *d =
-                vmaf_dictionary_merge(&rfe->fex_ctx[i]->opts_dict,
-                                      &fex_ctx->opts_dict);
-            vmaf_dictionary_free(&rfe->fex_ctx[i]->opts_dict);
-            rfe->fex_ctx[i]->opts_dict = d;
-            return vmaf_feature_extractor_context_destroy(fex_ctx);
+            /* same fex */
+            if (flags & VMAF_FEATURE_EXTRACTOR_CONTEXT_DO_NOT_OVERWRITE) {
```

Comment:
I'm pretty sure vmaf_dict_merge handles these cases you check for here. Should just be a matter of passing the correct flag.